Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -64,19 +64,32 @@ public IEnumerable<CourseComponentDTO> GetAllCourseComponents()
     public async Task<IEnumerable<string>> GetAllCourseSubjects()
     {
         return await _courseRepository.GetUniqueCourseSubjects();
+    }
+
+    public async Task<Course> GetCourseDataOrThrowOnDeleted(string subject, string catalog)
+    {
+        var course = await _courseRepository.GetCourseBySubjectAndCatalog(subject, catalog) 
+            ?? throw new BadRequestException($"The course {subject}-{catalog} does not exist.");
+
+        if (course.CourseState == CourseStateEnum.Deleted)
+        {
+            throw new BadRequestException($"The course {subject}-{catalog} is deleted.");
+        }
+
+        return course;

```

Comment:
I think it makes more sense for the exceptions to be `InvalidInputException` not `BadRequestException` at least for the first.